### PR TITLE
fix(css): update `!important` vs layers note

### LINKED ISSUE #16129: Ambiguous paragraph in “Introducing the CSS Cascade” doc
### MDN URL

https://developer.mozilla.org/en-US/docs/Web/CSS/Cascade

### What specific section or headline is this issue about?

The note about !important reversing the precedence of cascade layers

### What information was incorrect, unhelpful, or incomplete?

Note: !important reverses the precedence of cascade layers. For this reason, rather than using !important to override external styles, import frameworks, third party styles, widget stylesheets, etc., into layers, demoting their precedence. !important should only be used sparingly, if ever, to guard required styles against later overrides, in the first declared layer.

### What did you expect to see?

>For this reason, rather than using !important to override external styles, import frameworks, third party styles, widget stylesheets, etc., **(missing something here, group, move, transform, place)** into layers, demoting their precedence.

Not really sure what the best word is to make the paragraph obvious. Right now, this paragraph can lead to several interpretations. 

I think the author intended to say that **third parties** should be grouped/organised using layers, because of the mention of "their precedence", referring to third parties. But for this to be true, this note also implies that the author styles are declared outside a layer so they take precedence.

In my head, this is an example of what the author meant: 

```css
@import url(third-party.css) layer(thirdparty);

p {
  color: red;
}
```

but perhaps the ideal example would be (although this requires a bigger change in the note): 

```css

@layer thirdparty, base;

@import url(third-party.css) layer(thirdparty);

@layer base {
  p {
    color: red;
  }
}
```

Another option is to turn this note into a paragraph and provide a full example of how to set up layers to avoid using !important. 

### Do you have any supporting links, references, or citations?

_No response_

### Do you have anything more you want to share?

_No response_

### MDN metadata

<details>
<summary>Page report details</summary>

* Folder: `en-us/web/css/cascade`
* MDN URL: https://developer.mozilla.org/en-US/docs/Web/CSS/Cascade
* GitHub URL: https://github.com/mdn/content/blob/main/files/en-us/web/css/cascade/index.md
* Last commit: https://github.com/mdn/content/commit/f9861ad43f9b33ca9f1b6a4b1d403be5276763db
* Document last modified: 2022-05-16T17:26:04.000Z

</details>

### PR DESCRIPTION
**Pull request by OnkarRuikar**
- fix https://github.com/mdn/content/issues/16129

Let's break the statement. As the reporter says, it's not clear where the first clause ends.

### CODE REVIEW

File: files/en-us/web/css/cascade/index.md [hunk 1]
```diff
@@ -283,7 +283,7 @@ p {
...
 
-> **Note:** `!important` reverses the precedence of cascade layers. For this reason, rather than using `!important` to override external styles, import frameworks, third party styles, widget stylesheets, etc., into layers, demoting their precedence. `!important` should only be used sparingly, if ever, to guard required styles against later overrides, in the first declared layer.
+> **Note:** `!important` reverses the precedence of cascade layers. For this reason, try not to use `!important` to override external styles. Instead, import the external style sheets(from frameworks, widget stylesheets, libraries, etc.) into initial layers, which will demote their precedence, and author-defined layers will have higher precedence. `!important` should only be used sparingly, if ever, to guard required styles against later overrides, in the first declared layer.
```
Comment (estelle) [inline files/en-us/web/css/cascade/index.md:286]:
```suggestion
> **Note:** The `!important` flag reverses the precedence of cascade layers. For this reason, try not to use `!important` to override external styles. Instead, import external style sheets (from frameworks, widget stylesheets, libraries, etc.) into named layers. Importing stylesheets into a layer as the first declaration in your CSS demotes their precedence, and author-defined layers, defined later in your CSS, will have higher precedence. The `!important` flag should only be used sparingly, if ever, to guard required styles against later overrides, in the first declared layer.
```

`initial` wasn't clear. My suggestion is wordy, but i think clarifies it?

File: files/en-us/web/css/cascade/index.md [hunk 2]
```diff
@@ -343,7 +343,7 @@ Finally, {{cssxref("@charset")}} obeys specific algorithms and isn't affected by
...
 
-If the HTML presentation attribute is supported by the user agent, valid presentational attributes included in HTML are translated to the corresponding CSS rules, such as the {{cssxref("align")}} or `fill` property (all SVG presentation attributes can be used as a CSS property) inserted in the author stylesheet prior to any other styles with a specificity equal to `0`.
+If the HTML presentation attribute is supported by the user agent, valid presentational attributes included in HTML are translated to the corresponding CSS rules, such as the [`align`](/en-US/docs/Web/API/HTMLImageElement/align) or [`fill`](/en-US/docs/Web/SVG/Attribute/fill) property (all SVG presentation attributes can be used as a CSS property) inserted in the author stylesheet prior to any other styles with a specificity equal to `0`.
```
Comment (estelle) [inline files/en-us/web/css/cascade/index.md:346]: how about https://developer.mozilla.org/en-US/docs/Web/HTML/Element/img#align for align?
I plan on creating the fill css property this week. Not sure if it's premature to macro it? https://github.com/orgs/mdn/projects/25/views/1?pane=issue&itemId=71536919

**Review by estelle — COMMENTED** [review]
suggestions for you to consider

**Comment by Josh-Cena** [comment]
Does this fix #16129?

File: files/en-us/web/css/cascade/index.md (hunk 1 above)
Comment (estelle) [reply files/en-us/web/css/cascade/index.md:286]: created issue https://github.com/mdn/content/issues/35005

Comment (estelle) [reply files/en-us/web/css/cascade/index.md:286]: This edit will fix https://github.com/mdn/content/issues/16129

**Comment by estelle** [comment]
> Does this fix #16129?

Yes.

File: files/en-us/web/css/cascade/index.md (hunk 2 above)
Comment (OnkarRuikar) [reply files/en-us/web/css/cascade/index.md:346]:
```suggestion
If the HTML presentation attribute is supported by the user agent, valid presentational attributes included in HTML are translated to the corresponding CSS rules, such as the [`align`](/en-US/docs/Web/HTML/Element/img#align) or [`fill`](/en-US/docs/Web/SVG/Attribute/fill) property (all SVG presentation attributes can be used as a CSS property) inserted in the author stylesheet prior to any other styles with a specificity equal to `0`.
```

File: files/en-us/web/css/cascade/index.md [hunk 3]
```diff
@@ -343,7 +343,7 @@ Finally, {{cssxref("@charset")}} obeys specific algorithms and isn't affected by
...
 
-If the HTML presentation attribute is supported by the user agent, valid presentational attributes included in HTML are translated to the corresponding CSS rules, such as the {{cssxref("align")}} or `fill` property (all SVG presentation attributes can be used as a CSS property) inserted in the author stylesheet prior to any other styles with a specificity equal to `0`.
+If the HTML presentation attribute is supported by the user agent, valid presentational attributes included in HTML are translated to the corresponding CSS rules, such as the [`align`](/en-US/docs/Web/HTML/Element/img#align) or [`fill`](/en-US/docs/Web/SVG/Attribute/fill) property (all SVG presentation attributes can be used as a CSS property) inserted in the author stylesheet prior to any other styles with a specificity equal to `0`.
```
Comment (estelle) [inline files/en-us/web/css/cascade/index.md:346]:
```suggestion
If the HTML presentation attribute is supported by the user agent, valid presentational attributes included in HTML and SVG, such as the [`align`](/en-US/docs/Web/HTML/Element/img#align) or [`fill`](/en-US/docs/Web/SVG/Attribute/fill) attributes, are translated to the corresponding CSS rules (all SVG presentation attributes are supported as CSS properties) and inserted in the author stylesheet prior to any other styles with a specificity equal to `0`.
```

**Review by estelle — APPROVED** [review]
Thank you! 🎉

**Pull request merged**